packages/app-storage/src/Query.tsx

```diff
@@ -130,6 +135,58 @@ class Query extends React.PureComponent<Props, State> {
     );
   }
 
+  private typeIsTuple (type: string): boolean {
+    return type.charAt(0) === '(' && type.charAt(type.length - 1) === ')';
+  }
+
+  private breakDownTuple (type: string): Array<any> {
+    return type.substring(1, type.length - 1).split(',');
+  }
+
+  private renderInputs () {
+    const value = this.props.value;
+    const { key } = value;
+
+    // if function has inputs, render them
+    if ((value as StorageModuleQuery).params.length > 0) {
+      const type = isU8a(key) ? 'Data' : key.meta.type.asMap.key.toString();
+      const params = (value as StorageModuleQuery).params;
+      return (
+        <div>
+          Inputs: ({type})
+          {
+            params.map((obj) => {
```

Comment:
This would actually be better in valueToText itself so it is re-usable. Basically here we should be really light on logic and just pass the stuff we get in as-is for formatting. (In this app we are, sadly, already quite logic heavy I believe - blame me.)